src/main/java/hu/wurfel/refference/school/diary/DiaryController.java

```diff
@@ -0,0 +1,53 @@
+package hu.wurfel.refference.school.diary;
+
+import hu.wurfel.refference.school.model.AddRequest;
+import hu.wurfel.refference.school.student.Student;
+import hu.wurfel.refference.school.student.StudentRequest;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.ArrayList;
+
+@RestController
+public class DiaryController {
+
+	@Autowired
+    ResponseCreationClarificationDiary responseCreationClarificationDiary;
+
+	@Autowired
+    DiaryService diaryService;
+
+	@PostMapping("/api/v1/search/Student")
+	public ResponseEntity<ArrayList> searchForStudent(@RequestBody DiaryRequest diaryRequest) {
+		ArrayList answer = new ArrayList();
+		answer = responseCreationClarificationDiary.create(diaryRequest);
+		return ResponseEntity.ok(answer);
+	}
+
+	@PostMapping("/api/v1/adding/Student")
+	public ResponseEntity<ArrayList> adding (@RequestBody DiaryRequest diaryRequest){
+		ArrayList answer =new ArrayList();
+		answer.add(diaryService.saveDiary(Integer.parseInt(diaryRequest.getId()),Long.parseLong(diaryRequest.getStudentId()),Integer.parseInt(diaryRequest.getClassId())));
+		return  ResponseEntity.ok(answer);
+	}
+
+	@PostMapping("/api/v1/delete/Student")
```

Comment:
Mivel mindenre megy egy commitod és egyből merge-ölöd masterbe (ilyet nem csinálunk, főleg ha valaki review-za), ezért nem tudok suggestiont beszúrni, mert csak pull requesthez (PR) lehet.

@DeleteMapping("/api/v1/students/{id}")
public void deleteById(@PathVariable String id)  // nyilván mehet más típus is - a string id csak akkor célszerű, ha te generálod, amúgy longot használj

@GetMapping - lekérés, nincs request body
@PostMapping - alapvetően ez a create, de a kivételeket leszámítva akkor haszáljuk, ha van érkező request body
@DeleteMapping - ez elég egyértelmű
@PutMapping - ez az update, ennél a teljes entitás kell, ez vagy teljes vagy majdnem teljes update (nyilván az egyedi azonosító és hasonlók nem kerülnek frissítésre
@PatchMapping - ez a részleges update, pl van egy studented, aki átmegy másik suliba és jelezni szeretnéd az iskola váltást, akkor az adott student id-jával átállítod az iskola mezőt kb így

@PatchMapping("/api/v1/students/{id}/schools/{schoolId}")
public void changeSchool(@PathVariable String id, @PathVariable String schoolId)

Ezt akkor csinálnám, ha a studentnél van mentve, hogy melyik iskolához tartozik.
Ha az iskolánál van mentve, akkor a SchoolControllerben csinálnám.

Reply:
A nagybetű még maradt az Enumban mentett Entity nevek miatt, de egyenlőre jelentésen átalakítottam, és szerintem mindenhova az oda illő Mapping került